proto/site/v1alpha/site.proto

```diff
@@ -0,0 +1,252 @@
+message DeleteMemberRequest {
+  // Required. Mintter Account ID of the member to be deleted.
+  string account_id = 1;
+}
+
+// Request to list blocked accounts.
+message ListBlockedAccountsRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Options. Token for a specific page if any.
+  string next_page_token = 2;
+}
+
+// Response with a list of blocked accounts.
+message ListBlockedAccountsResponse {
+  // List of blocked accounts.
+  repeated string accounts = 1;
+
+  // Token for the next page if any.
+  string next_page_token = 2;
+}
+
+// Invite token is produced by the owner of the site to invite new members.
+message InviteToken {
+  // The value of the invite token.
+  string token = 1;
+
+  // Timestamp after which the token will not be recognized by the site.
+  google.protobuf.Timestamp expire_time = 2;
+}
+
+// Public-facing information about the Mintter Site.
+message SiteInfo {
+  // Hostname under which this site is exposed.
+  string hostname = 1;
+
+  // Title of this site.
+  string title = 2;
+
+  // Description of this site.
+  string description = 3;
+
+  // Mintter Account ID that is configured as the owner of this site.
+  string owner = 4;
+}
+
+// Member of the site.
+message Member {
+  // Roles a member of the Mintter Site can have.
+  enum Role {
+    // Invalid default value.
+    ROLE_UNSPECIFIED = 0;
+
+    // Currently there can only be one owner of the site.
+    OWNER = 1;
+
+    // Editors are allowed to push content to the site.
+    EDITOR = 2;
+  }
+
+  // Mintter Account ID of the member.
+  string account_id = 1;
+
+  // Member's role on the site.
+  Role role = 2;
+}
+
+// Response for the /.well-known discovery HTTP page.
+// TODO: define the actual /.well-known path.
+message DiscoveryConfig {
+  // TODO: add the necessary fields to export under the /.well-known HTTP discovery path.
+}
+
+// Request the server to pin+publish a document.
+message PublishRequest {
+  // id of the doc to publish
+  string doc_id = 1;
+
+  // (optional) id of the web publication that is being updated. If empty, a new publication is being created.
+  string publication_id = 2;
+
+  // (optional) pretty path to publish at. Empty string === home doc.
+  string path = 3;
+
+  // version of the doc to publish
+  string version = 4;
+
+  // list of doc ids that are referenced by this doc via links or transclusions
+  repeated string referenced_doc_ids = 5;
+}
+
+// Response when publishing a document.
+message PublishResponse {
+  // id of the web publication on this server
+  string publication_id = 1;
+}
+
+// Request Unpin/Unpublish a document from the site.
+message UnpublishRequest {
+  // id of web publication on this server to unpublish
+  string publication_id = 1;
+}
+
+message UnpublishResponse {}
+
+// Object representing a "web publication" row on this server, including fields for user presentation (doc_title, author_name, update_time)
+message ListedWebPublication {
```

Comment:
I'd name this `WebPublication`, or even just `Publication`.

Reply:
Publication is taken. Wouldn't the representation be different depending on the context in which it is listed? For example see `GetDocWebPublications` in `sites.proto` which presents the WebPub in a different way (including hostname)